### PR TITLE
include tags on nodemanager process node

### PR DESCRIPTION
**Pull request by vjeffrey**
### :nut_and_bolt: Description
this change is in support of the work towards the
one node view.  also included is a change to ensure we only
add a tag if it does not already exist in the db and only
add a node id-tag id association if it does not already exist

### :+1: Definition of Done
tags are passed through from scan job -> report
tags are added as part of the processNode codepath
tags are not added to the db if they already exist in the system
node tags associations are not added to the db if they already exist in the system

### :chains: Related Resources
https://github.com/chef/automate/issues/192

### CODE REVIEW

File: components/nodemanager-service/pgdb/node-managers-process-node.go [hunk 1]
```diff
@@ -60,16 +60,31 @@ func (db *DB) ProcessIncomingNode(node *manager.NodeMetadata) error {
...
-		_, err = db.Exec(sqlUpsertByID, node.GetUuid(),
-			node.GetName(), node.GetPlatformName(), node.GetPlatformRelease(),
-			nodeState, lastContact, node.GetSourceRegion(), node.GetSourceAccountId(),
-			node.GetJobUuid())
-	} else {
-		_, err = db.Exec(sqlUpsertBySourceID, node.GetUuid(),
-			node.GetName(), node.GetPlatformName(), node.GetPlatformRelease(),
-			nodeState, lastContact, node.GetSourceId(), node.GetSourceRegion(),
-			node.GetSourceAccountId(), node.GetJobUuid())
+	err = Transact(db, func(tx *DBTrans) error {
```
Comment (lancewf) [inline components/nodemanager-service/pgdb/node-managers-process-node.go]: I like the Transact here.

File: components/compliance-service/inspec-agent/types/types.go [hunk 2]
```diff
@@ -67,6 +68,7 @@ type InspecJob struct {
 	ProfilesOwner     string              `json:"profiles_owner,omitempty"`
 	InternalProfiles  []string            `json:"internal_profiles,omitempty"`
 	MachineIdentifier string              `json:"machine_identifier,omitempty"`
+	Tags              []*common.Kv        `json:"kv,omitempty"`
```
[the remark this answers is not in the document]
Comment (vjeffrey) [reply components/compliance-service/inspec-agent/types/types.go]: oh ya it totally should!

File: components/nodemanager-service/pgdb/tags.go [hunk 3]
```diff
@@ -29,15 +50,56 @@ func (trans *DBTrans) addTags(tags []*common.Kv) ([]string, error) {
...
+			link := NodeTag{
+				NodeID: nodeID,
+				TagID:  tagID,
+			}
+			links = append(links, &link)
 		}
-		links = append(links, &link)
 	}
 	return trans.Insert(links...)
 }
```
Comment (phiggins) [inline components/nodemanager-service/pgdb/tags.go]: I realize this is asking for quite a bit of rework, but I feel like this database syncing stuff would be better done in the database, by making the relevant columns unique and doing `INSERT ... ON CONFLICT DO NOTHING`.

**Comment by phiggins** [comment]
```
CONFIG-008

One of the aliases you have provided in project.alias is already in use by another agent. Please update your list to remove the conflict.
```

The CI failure is probably not related to this PR and I'm guessing just needs a retry, but I've never seen this one before.

**Comment by vjeffrey** [comment]
I think the CI failure is b/c i'm branched off a branch

File: components/nodemanager-service/pgdb/tags.go (hunk 3 above)
Comment (vjeffrey) [reply components/nodemanager-service/pgdb/tags.go]: ah good point, that would be cleaner. i'll give that a try.

**Comment by vjeffrey** [comment]
that's so annoying. i went to change the base branch and it closed my pr and then told me i can't change the base branch on a closed pr :/
i think it was closed because the base branch was deleted. bleh

**Comment by vjeffrey** [comment]
well anyways, ill work on the improvements mentioned and the reopen with a new pull request since it won't allow me to re-open this one

**Pull request closed without merging**